### ISSUE TITLE
When using [routerLink] to construct a link that updates the primary and a named outlet, a trailing slash is added making the URL invalid

### ISSUE THREAD

**Issue by plex4r** [angular/angular#38469]
# 🐞 bug report

 The issue is caused by package @angular/router

### Is this a regression?

 No

### Description

When I construct a link that needs to update the primary and a secondary outlet using [routerLink] the final URL contains an additional '/' that causes Angular to consider it invalid.

The link is constructed like this:   <a  [routerLink]="['/uw/tagent/', myid,  { outlets: { center: ['gi', myid] } }  ]" >
This results in the following URL:    http://localhost:8080/uw/tagent/9980/(center:gi/9980)
This throws the following error:  Error: Cannot match any routes. ... (please see full stack below)

This is not valid because of the '/' after myid which is '9980' in my example.

What I need is a way to construct the URL: http://localhost:8080/uw/tagent/9980(center:gi/9980)
This link works perfectly.  

I believe this issue has been raised before, or something similar to it such as this: https://github.com/angular/angular/issues/21003  

But the resolution what that it was not considered a bug.
None of the workarounds I have found will work for me.  
One such workaround is to create a function that constructs the link, and a link that points to it.  The function then navigates by this.router.navigateByUrl(x)...
However, I do not get the benefit of the link being highlighted after it is clicked like [routerLink] provides in html.

I believe this is a bug because I am using Angular to construct a link that is later says is invalid.  If it is not a bug, can you please provided a workaround where I can construct a valid link using [routerLink].

Thank you for the help.

## 🔬 Minimal Reproduction

Use my code above to construct an invalid link.

## 🔥 Exception or Error
<pre><code>

vendor-es2015.js:32579 ERROR Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'uw/tagent/9980'
Error: Cannot match any routes. URL Segment: 'uw/tagent/9980'
    at ApplyRedirects.noMatchError (vendor-es2015.js:86697)
    at CatchSubscriber.selector (vendor-es2015.js:86661)
    at CatchSubscriber.error (vendor-es2015.js:118921)
    at MapSubscriber._error (vendor-es2015.js:116181)
    at MapSubscriber.error (vendor-es2015.js:116161)
    at MapSubscriber._error (vendor-es2015.js:116181)
    at MapSubscriber.error (vendor-es2015.js:116161)
    at MapSubscriber._error (vendor-es2015.js:116181)
    at MapSubscriber.error (vendor-es2015.js:116161)
    at ThrowIfEmptySubscriber._error (vendor-es2015.js:116181)
    at resolvePromise (polyfills-es2015.js:1271)
    at resolvePromise (polyfills-es2015.js:1223)
    at polyfills-es2015.js:1333
    at ZoneDelegate.invokeTask (polyfills-es2015.js:872)
    at Object.onInvokeTask (vendor-es2015.js:64137)
    at ZoneDelegate.invokeTask (polyfills-es2015.js:871)
    at Zone.runTask (polyfills-es2015.js:640)
    at drainMicroTaskQueue (polyfills-es2015.js:1042)

</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 9.1.0
Node: 12.3.1
OS: win32 x64

Angular: 9.1.1
... animations, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.0
@angular-devkit/build-angular     0.901.0
@angular-devkit/build-optimizer   0.901.0
@angular-devkit/build-webpack     0.901.0
@angular-devkit/core              9.1.0
@angular-devkit/schematics        9.1.0
@angular/cdk                      9.2.0
@angular/cli                      9.1.0
@ngtools/webpack                  9.1.0
@schematics/angular               9.1.0
@schematics/update                0.901.0
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0

</code></pre>

**Anything else relevant?**

I do not believe this is browser specific since Angular is throwing the error.

OS does not matter

**Comment by atscott** [comment]
Hi @plex4r, can you please provide a reproduction in stackblitz for this so we can investigate?

**Comment by atscott** [comment]
Here's a repro: https://stackblitz.com/edit/angular-auxiliary-routes-dk3lva?file=app%2Fapp.component.html

Can you use `<a [routerLink]="[{ outlets: { primary: ['uw' 'tagent', myid], center: ['gi', myid] } }]" >` as a workaround? Remember not to use slashes in the commands there (#13872).

There are [many issues with auxiliary routes](https://github.com/angular/angular/labels/router%3A%20aux%20routes) and I'm surprised none of them cover this scenario. #13523 does seem like it might be the same as this, just described in a different way. https://github.com/angular/angular/issues/13523#issuecomment-267692567 and https://github.com/angular/angular/issues/13523#issuecomment-267696029 describe similar workarounds.

**Comment by localcrafts101** [comment]
**anggg@![image](https://user-images.githubusercontent.com/53924846/90305982-ef041a80-de8d-11ea-8e0e-282273250674.png)**

**Comment by plex4r** [comment]
Hey atscott,
Thank you for the Stackblitz example.  It does cleanly reproduce the issue and it helped me with a workaround.  I had to also clear the primary because it was duplicating the primary link in the parenthesis.  Here is what I am using:
 `<a [routerLink]="['', { outlets: { primary: ['uw' 'tagent', myid], center: ['gi', myid] } }]" >`

Thank you very much for the help.
I hope this does get schedule for a fix since someone else will probably run into this same issue.

Regards.

**Comment by leandroz** [comment]
I have an issue that I think is related.

Routes definition
```
const routes: Routes = [
  { path: '',   redirectTo: '/login', pathMatch: 'full' },
  {
    path: 'login',
    component: LoginComponent,
    canActivate: [NotLoggedInGuard]
  },
  {
    path: 'sign-up',
    component: SignUpComponent,
    canActivate: [NotLoggedInGuard]
  },
  {
    path: 'authorize',
    component: AuthorizeComponent,
    canActivate: [LoggedInGuard]
  },
  {
    path: 'app',
    component: MyAccountComponent,
    canActivate: [LoggedInGuard],
    runGuardsAndResolvers: 'always',
    children: [
      { path: '',   redirectTo: '/app/(tickets//content:default)', pathMatch: 'full' },
      {
        path: 'settings',
        component: SettingsComponent,
      },
      {
        path: 'my-profile',
        component: MyProfileComponent,
        outlet: 'content'
      },
      {
        path: 'statistics',
        component: StatisticsComponent,
      },
      {
        path: 'analytics',
        component: AnalyticsComponent,
        outlet: 'content'
      },
      {
        path: 'tickets',
        component: TicketViewsComponent,
      },
      {
        path: 'views/:viewId',
        component: TicketsComponent,
        outlet: 'content',
      },
      {
        path: 'customers',
        component: CustomerViewsComponent,
      },
      {
        path: 'customer-views/:viewId',
        component: CustomersComponent,
        outlet: 'content',
      },
      {
        path: 'default',
        component: NoContentComponent,
        outlet: 'content'
      },
    ]
  },
  { path: '**', redirectTo: '/login', pathMatch: 'full' },
];
```
MyAccountComponent template
```
<mat-sidenav-container>
  <mat-sidenav>

    <mat-list class="header"></mat-list>

    <router-outlet class="views"></router-outlet>

    <mat-action-list dense class="footer"></mat-action-list>
  </mat-sidenav>

  <mat-sidenav-content class="main-container">
    <router-outlet name="content"></router-outlet>
  </mat-sidenav-content>

</mat-sidenav-container>
```

The url generated to correctly navigate to let's say the TicketsViewComponent in primary and TicketsComponent in content named outlet is `/app/(tickets//content:views/0001)`, note the double `/` between tickets and content.
The problem is that is redirected as just one in Firebase Hosting and I believe probably the same in other providers.

To solve the issue I am manually replacing /content for //content when appropriate, but I believe is a bug.

```
export class CustomUrlSerializer implements UrlSerializer {
  defaultUrlSerializer = new DefaultUrlSerializer();
  parse(url: string): UrlTree {
    if (!url.includes('//content') && url.includes('/content')) {
      url = url.replace('/content', '//content');
    }
    return this.defaultUrlSerializer.parse(url);
  }

  serialize(tree: UrlTree): string {
    return this.defaultUrlSerializer.serialize(tree);
  }
}
```

**Comment by plex4r** [comment]
leandroz, that does look related but I am not sure if that is the result of the same root issue.  I assume you are using something like Router.navigateByUrl after you correct the URL.  I just wanted to comment that I saw that same behavior in constructing the URL using [routerLink] in the HTML template, in one of my multiple failed attempts to correct the extra slash before the '(' (which was the issue I kept having).
